app/services/flood_risk_engine/simple_encoding.rb

```diff
@@ -0,0 +1,26 @@
+module FloodRiskEngine
+  # Simple cipher encoding from
+  # http://stackoverflow.com/questions/4128939/simple-encryption-in-ruby-without-external-gems
+  module SimpleEncoding
+
+    protected
+
+    def encode(text)
+      return "" unless text.present?
+      text.tr(characters, encoding)
+    end
+
+    def decode(text)
+      return "" unless text.present?
+      text.tr(encoding, characters)
+    end
+
+    def characters
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
+    end
+
+    def encoding
+      "N8nGfS1rpcHvmoiP6WB9FMQwXjyZ0Rhu73stKAla2Yd4LTJqkeUzgEOxbCD5VI"
+    end
+  end
+end
```

Comment:
Youve probably looked at this but you can use a signed cookie which will automatically sign it using secret key base - though actually I'm not sure if its actually encrypted too, haven't checked
from http://api.rubyonrails.org/classes/ActionDispatch/Cookies.html
```
# Sets a signed cookie, which prevents users from tampering with its value.
# The cookie is signed by your app's `secrets.secret_key_base` value.
# It can be read using the signed method `cookies.signed[:name]`
cookies.signed[:user_id] = current_user.id
```